source/extensions/filters/network/rbac/rbac_filter.cc

```diff
@@ -0,0 +1,82 @@
+#include "extensions/filters/network/rbac/rbac_filter.h"
+
+#include "envoy/buffer/buffer.h"
+#include "envoy/network/connection.h"
+
+#include "extensions/filters/network/well_known_names.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace RBACFilter {
+
+RoleBasedAccessControlFilterConfig::RoleBasedAccessControlFilterConfig(
+    const envoy::config::filter::network::rbac::v2::RBAC& proto_config, Stats::Scope& scope)
+    : stats_(Filters::Common::RBAC::generateStats(proto_config.stat_prefix(), scope)),
+      engine_(Filters::Common::RBAC::createEngine(proto_config)),
+      shadow_engine_(Filters::Common::RBAC::createShadowEngine(proto_config)) {}
+
+Network::FilterStatus RoleBasedAccessControlFilter::onData(Buffer::Instance&, bool) {
+  ENVOY_LOG(
+      debug, "checking connection: remoteAddress: {}, localAddress: {}, ssl: {}",
+      callbacks_->connection().remoteAddress()->asString(),
+      callbacks_->connection().localAddress()->asString(),
+      callbacks_->connection().ssl()
+          ? "uriSanPeerCertificate: " + callbacks_->connection().ssl()->uriSanPeerCertificate() +
+                ", subjectPeerCertificate: " +
+                callbacks_->connection().ssl()->subjectPeerCertificate()
+          : "none");
+
+  if (shadow_engine_result_ == Unknown) {
+    // TODO(quanlin): Support metric collection for RBAC network filter.
```

Comment:
Don't you check metrics below? Is this comment still valid?

Reply:
Ah, sorry the comment might be a bit misleading, I actually mean to pass the metrics to other filers. In HTTP filter this is done by using metadata. But there is no metadata for TCP filter so I created this TODO.